src/test/java/com/example/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,284 @@
+        expectedBookDto.setCoverImage("image1.jpg");
+        expectedBookDto.setCategoriesIds(Set.of(1L));
+
+        expectedBookDto2 = new BookDto();
+        expectedBookDto2.setId(2L);
+        expectedBookDto2.setTitle("Book 2");
+        expectedBookDto2.setAuthor("Author 2");
+        expectedBookDto2.setIsbn("ISBN-002");
+        expectedBookDto2.setPrice(BigDecimal.valueOf(30));
+        expectedBookDto2.setDescription("Description for Book 2");
+        expectedBookDto2.setCoverImage("image2.jpg");
+        expectedBookDto2.setCategoriesIds(Set.of(2L));
+
+        createBookRequest = new CreateBookRequestDto();
+        createBookRequest.setTitle("Pride and Prejudice");
+        createBookRequest.setAuthor("Jane Austen");
+        createBookRequest.setIsbn("978-1-60309-515-0");
+        createBookRequest.setPrice(new BigDecimal(13));
+        createBookRequest.setDescription("A classic romance novel");
+        createBookRequest.setCoverImage("https://pride.com/prejudice.jpg");
+        Set<Long> categoryIdsForDto = new HashSet<>();
+        categoryIdsForDto.add(2L);
+        createBookRequest.setCategoryIds(categoryIdsForDto);
+
+        expectedCreatedBookDto = new BookDto();
+        expectedCreatedBookDto.setTitle(createBookRequest.getTitle());
+        expectedCreatedBookDto.setAuthor(createBookRequest.getAuthor());
+        expectedCreatedBookDto.setIsbn(createBookRequest.getIsbn());
+        expectedCreatedBookDto.setCoverImage(createBookRequest.getCoverImage());
+        expectedCreatedBookDto.setDescription(createBookRequest.getDescription());
+        expectedCreatedBookDto.setCategoriesIds(categoryIdsForDto);
+        expectedCreatedBookDto.setPrice(createBookRequest.getPrice());
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin")
+    @DisplayName("findAll_returnsList_ok")
+    @Test
+    @Order(0)
+    void findAll_returnsList_ok() {
+        MvcResult result = mockMvc.perform(get("/api/books?page=0&size=2")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().is2xxSuccessful()).andReturn();
+        List<BookDto> actual =
+                objectMapper.readValue(result.getResponse().getContentAsString(), ArrayList.class);
+        List<BookDto> actualDtos = objectMapper.convertValue(actual,
+                new TypeReference<List<BookDto>>() {
+                });
+        int expectedSize = 2;
+        assertEquals(expectedSize, actual.size());
+
+        boolean isEquals = EqualsBuilder.reflectionEquals(
+                expectedBookDto, actualDtos.get(0), "id");
+        boolean isEqualsSecond = EqualsBuilder.reflectionEquals(
+                expectedBookDto2, actualDtos.get(1), "id");
+        assertTrue(isEquals);
+        assertTrue(isEqualsSecond);
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @DisplayName("createBook_returnsCreatedBook_ok")
```

Comment:
we add this annotation to add a more clear description of test cases. It doesn't make sense if you just write the name of the method. For example, `@DisplayName("Test createBook with a valid request")`. Please, do it with every method